### ISSUE TITLE
ios_user tries to set a user's SSH key-hash multiple times

### ISSUE THREAD

**Issue by berrym-akrr** [ansible-collections/cisco.ios#920]
##### SUMMARY
When using ios_user with the sshkey option, the module sends the command to install the key 3 times if the key does not exist, and twice if the user and key are already on device. This is seemingly harmless, but does result in the task being marked as changed in the second case.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
cisco.ios.ios_user

##### ANSIBLE VERSION

```paste below
ansible [core 2.15.3]
  config file = None
  configured module search path = ['/home/matthew/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/matthew/Desktop/Ansible/AnsibleEnv/lib/python3.11/site-packages/ansible
  ansible collection location = /home/matthew/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/matthew/Desktop/Ansible/AnsibleEnv/bin/ansible
  python version = 3.11.4 (main, Jun  9 2023, 07:59:55) [GCC 12.3.0] (/home/matthew/Desktop/Ansible/AnsibleEnv/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION

```paste below
# /home/matthew/Desktop/Ansible/AnsibleEnv/lib/python3.11/site-packages/ansible_collections
Collection Version
---------- -------
cisco.ios  4.6.1
```

##### CONFIGURATION

```paste below
CONFIG_FILE() = None
DEFAULT_HOST_LIST(env: ANSIBLE_INVENTORY) = ['/home/matthew/Desktop/Ansible/AnsibleNetwork/inventory']
```

##### OS / ENVIRONMENT

Router#show ver
Cisco IOS Software, C3560C Software (C3560c405ex-UNIVERSALK9-M), Version 15.2(2)E10, RELEASE SOFTWARE (fc2)
Technical Support: http://www.cisco.com/techsupport
Copyright (c) 1986-2019 by Cisco Systems, Inc.
Compiled Tue 14-May-19 10:28 by prod_rel_team

##### STEPS TO REPRODUCE

```yaml
    - name: Test user
      cisco.ios.ios_user:
        aggregate:
          - name: test
            configured_password: "{{ lookup('community.general.random_string', length=128, special=false) }}"
            sshkey: "{{ lookup('file', 'id_rsa_example.pub') }}"
        privilege: 15
        password_type: secret
        update_password: on_create
```

##### EXPECTED RESULTS

On the first playbook run, the module adds the user and key once and only once.
On the second playbook run, the module reports no change

##### ACTUAL RESULTS

First run:
```paste below
changed: [192.168.117.16] => {
    "changed": true,
    "commands": [
        "username test privilege 15",
        "ip ssh pubkey-chain",
        "username test",
        "key-hash ssh-rsa BC919209F323D6A3631348C403C1A8A4",
        "exit",
        "exit",
        "username test privilege 15",
        "ip ssh pubkey-chain",
        "username test",
        "key-hash ssh-rsa BC919209F323D6A3631348C403C1A8A4",
        "exit",
        "exit",
        "username test privilege 15",
        "ip ssh pubkey-chain",
        "username test",
        "key-hash ssh-rsa BC919209F323D6A3631348C403C1A8A4",
        "exit",
        "exit"
    ],

```

Second run:
```
changed: [192.168.117.16] => {
    "changed": true,
    "commands": [
        "ip ssh pubkey-chain",
        "username test",
        "key-hash ssh-rsa BC919209F323D6A3631348C403C1A8A4",
        "exit",
        "exit",
        "ip ssh pubkey-chain",
        "username test",
        "key-hash ssh-rsa BC919209F323D6A3631348C403C1A8A4",
        "exit",
        "exit"
    ],

```

**Comment by berrym-akrr** [comment]
Ok, am I blind? I specified a password in the task, and I don't see the module running 'username test secret abcd'. The password for that user was indeed left blank. There seem the be differences in behavior depending on other users present on the device? Having the username 'test' being listed under ip ssh pubkey-chain with no matching user defined (username test priv 15 not in the config) seems to break the module in particular.

**Comment by roverflow** [comment]
> Ok, am I blind? I specified a password in the task, and I don't see the module running 'username test secret abcd'. The password for that user was indeed left blank. There seem the be differences in behavior depending on other users present on the device? Having the username 'test' being listed under ip ssh pubkey-chain with no matching user defined (username test priv 15 not in the config) seems to break the module in particular.

Could you share the play that you ran!

**Comment by berrym-akrr** [comment]
The playbook in question is the same one in the bug report above, I'm 99% certain I did not have a user named test on my lab device on the first run. You can see in the first run where the module sends the 'username test priv 15' command to create the user, but never sends 'username test secret abcd'.

**Comment by berrym-akrr** [comment]
I'm mainly setting both a ssh key and a random impossible password because if telnet gets left open, telnet will let an ssh-key user login with a blank password. It shouldn't ever happen, but if it does, that random password should prevent that issue.

Got it to reproduce that way, I really should have shown the before&after configs, sorry about that.
The key here to trigger the missing password is that the user test has a key listed under ip ssh pubkey-chain, but the user itself is NOT defined. I would guess that this makes on_create not fire.

Before:
```
Switch#show run | inc username
username ansible privilege 15 secret [removed]
# Username test NOT present

Switch#show run | sec ip ssh pubkey
ip ssh pubkey-chain
  username test
   key-hash ssh-rsa B3BF21239E5F4BFBC0915890DEEF3C89 
  username ansible
   key-hash ssh-rsa [removed]
```

Playbook as above
```
changed: [192.168.117.17] => {
    "changed": true,
    "commands": [
        "username test privilege 15",
        "ip ssh pubkey-chain",
        "username test",
        "key-hash ssh-rsa B3BF21239E5F4BFBC0915890DEEF3C89",
        "exit",
        "exit",
        "username test privilege 15",
        "ip ssh pubkey-chain",
        "username test",
        "key-hash ssh-rsa B3BF21239E5F4BFBC0915890DEEF3C89",
        "exit",
        "exit",
        "username test privilege 15",
        "ip ssh pubkey-chain",
        "username test",
        "key-hash ssh-rsa B3BF21239E5F4BFBC0915890DEEF3C89",
        "exit",
        "exit"
    ],
```

After:
```
Switch#show run | inc username
username ansible privilege 15 secret [removed]
username test privilege 15

Switch#show run | sec ip ssh pubkey
ip ssh pubkey-chain
  username test
   key-hash ssh-rsa B3BF21239E5F4BFBC0915890DEEF3C89 
  username ansible
   key-hash ssh-rsa [removed]
```

Reset to before config, run the playbook again with update_password: always
```
changed: [192.168.117.17] => {
    "changed": true,
    "commands": [
        "username test privilege 15",
        "ip ssh pubkey-chain",
        "username test",
        "key-hash ssh-rsa B3BF21239E5F4BFBC0915890DEEF3C89",
        "exit",
        "exit",
        "username test secret ********",
        "username test privilege 15",
        "ip ssh pubkey-chain",
        "username test",
        "key-hash ssh-rsa B3BF21239E5F4BFBC0915890DEEF3C89",
        "exit",
        "exit",
        "username test secret ********",
        "username test privilege 15",
        "ip ssh pubkey-chain",
        "username test",
        "key-hash ssh-rsa B3BF21239E5F4BFBC0915890DEEF3C89",
        "exit",
        "exit",
        "username test secret ********"
    ],
```

**Comment by roverflow** [comment]
@berrym-akrr  Yes, because the user is included in the 'ip ssh pubkey-chain,' it won't activate. One potential approach is to establish a new task dedicated to conducting this check using a distinct module and configure it accordingly. Typically, both 'configured_password' and 'sshkey' are not concurrently employed; hence, this unintended behavior
